lighthouse-core/runner.js

```diff
@@ -163,17 +165,22 @@ class Runner {
    * Resolves the location of the specified plugin and returns an absolute
    * string path to the file. Used for loading custom audits and gatherers.
    * Throws an error if no plugin is found.
-   * @param {string} plugin
+   * @param {(string|Gatherer|Audit)} plugin
```

Comment:
I think it would be preferable to leave `resolvePlugin` as strictly `string` input and output, roughly analogous to `require.resolve`.

If you take the code you're adding to `requireAudits` in config.js and also add it to `getGathererClass` in gather-runner.js, the plugin class will never make it this far, as those two spots can handle that case, and this can stay strings-only